pkg/zip_agnostic/reader_test.go

```diff
@@ -0,0 +1,66 @@
+package zip_agnostic
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestReader(t *testing.T) {
+	lorem := filepath.Join("testdata", "lorem.txt.golden")
+	want, err := os.ReadFile(lorem)
+	if err != nil {
+		t.Fatalf("can't read original file: %s", err)
+	}
+
+	fnames := []string{
+		lorem,
+		lorem + ".zst",
+		lorem + ".gz",
+	}
+
+	for _, fname := range fnames {
+		t.Run(fname, func(t *testing.T) {
+			f, err := os.Open(fname)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			r, err := NewReader(f)
+			if err != nil {
+				t.Fatalf("zagnosticReader returns %v", err)
```

Comment:
is this a typo?
```suggestion
				t.Fatalf("ZipAgnosticReader returns %v", err)
```

Reply:
good catch, it was the previous name :-)